R/utils-shared.R

```diff
@@ -226,12 +226,27 @@ splitDataGaps <- function(data, ts){
   
   data_list <- data[[ts$field[1]]]
   dataSplit <- list()
+  
+  hasGaps <- "gaps"  %in% names(data_list) && !isEmptyOrBlank(data_list$gaps)
+  hasEstimatedRangesAsGaps <- !isEmptyOrBlank(ts$estimated) && !ts$estimated == FALSE && 
+    "estimatedPeriods"  %in% names(data_list) && 
+    !isEmptyOrBlank(data_list$estimatedPeriods)
```

Comment:
logic for `hasEstimatedRangesAsGaps` will only apply to actual estimated timeseries. We need the stat timeseries that has estimated time periods. I think we need:

```r
hasEstimatedRangesAsGaps <- !isEmptyOrBlank(ts$estimated) && !ts$estimated && 
      "estimatedPeriods"  %in% names(data_list) && 
      !isEmptyOrBlank(data_list$estimatedPeriods)
```

This then is true for timeseries where ts$estimated is FALSE (so a non-estimated ts), there is a field for an estimated timeperiod in the data, and the estimated dates exist.